writeaheadlog.go

```diff
@@ -299,13 +255,10 @@ func (w *WAL) recoverWal(data []byte) ([]Update, error) {
 
 // writeRecoveryState is a helper function that changes the recoveryState on disk
 func (w *WAL) writeRecoveryState(state uint16) error {
```

Comment:
I think we don't need `writeRecoveryState`. If we change `writeWALMetadata` to take the state as an argument it should be as fast and we don't need the an additional function.

Reply:
will do

Reply:
Actually, going to do that in a follow-up pull request, imo it's a separate concern.